packages/table/test/tableTests.tsx

```diff
@@ -383,6 +387,149 @@ describe("<Table>", () => {
+                expect(component.state("viewportRect").left).to.equal(EXPECTED_LEFT_OFFSET);
+                component.simulate("keyDown", keyEventConfig);
+                expect(component.state("viewportRect").left).to.equal(EXPECTED_LEFT_OFFSET);
+            });
+
+            function runFocusCellViewportScrollTest(key: string,
+                                                    keyCode: number,
+                                                    attrToCheck: "top" | "left",
+                                                    expectedOffset: number) {
+                it(key, () => {
+                    const { component } = mountTable();
+                    component.simulate("keyDown", createKeyEventConfig(component, key, keyCode));
+                    expect(component.state("viewportRect")[attrToCheck]).to.equal(expectedOffset);
+                });
+            }
+        });
+
+
+        function mountTable(rowHeight = ROW_HEIGHT, colWidth = COL_WIDTH) {
+            const attachTo = document.createElement("div");
+            // need to `.fill` with some explicit value so that mapping will work, apparently
+            const columns = Array(NUM_COLS).fill(undefined).map((_, i) => <Column key={i} renderCell={renderCell}/>);
+            const component = mount(
+                <Table
+                    columnWidths={Array(NUM_ROWS).fill(colWidth)}
+                    enableFocus={true}
+                    focusedCell={DEFAULT_FOCUSED_CELL_COORDS}
+                    onFocus={onFocus}
+                    rowHeights={Array(NUM_ROWS).fill(rowHeight)}
+                    numRows={NUM_ROWS}
+                >
+                    {columns}
+                </Table>
+            , { attachTo });
+
+            // center the viewport on the focused cell
+            const viewportLeft = DEFAULT_FOCUSED_CELL_COORDS.col * COL_WIDTH;
+            const viewportTop = DEFAULT_FOCUSED_CELL_COORDS.row * ROW_HEIGHT;
+            const viewportWidth = COL_WIDTH;
+            const viewportHeight = ROW_HEIGHT;
+            component.setState({ viewportRect: new Rect(viewportLeft, viewportTop, viewportWidth, viewportHeight) });
+
+            return { attachTo, component };
+        }
+
+        function createKeyEventConfig(component: ReactWrapper<any, any>,
+                                      key: string,
+                                      keyCode: number,
+                                      shiftKey = false) {
+            const eventConfig = {
+                key,
+                keyCode,
+                shiftKey,
+                preventDefault: () => { /* Empty */ },
+                stopPropagation: () => { /* Empty */ },
+                target: (component as any).getNode(), // `getNode` is a real Enzyme method, just not in the typings?
+                which: keyCode,
+            };
+            return {
+                eventConfig,
+                nativeEvent: eventConfig,
```

Comment:
`HotkeysTarget` passes `event.nativeEvent` to the component's `onKeyDown`, so you have to explicitly override that object here (while also setting the top-level `key`, `which`, etc., because those are referenced here and there.)